File: PS/Programmers/더 맵게.cc
#include <iostream>
#include <queue>
#include <vector>
#include <cstdio>
#include <utility>
#include <cstring>
#include <string>
#include <set>
#include <algorithm>
using namespace::std;
int solution(vector<int> scoville, int K) {
	int answer = 0;
	priority_queue<int, vector<int>, greater<int>> pq;

	for (int i = 0; i < scoville.size(); i++)
	{
		pq.push(scoville[i]);
	}
	while (1)
	{
		if (pq.top() >= K)
			break;
		else if (pq.size() == 1 || pq.empty())
			return -1;
		int a = pq.top();
		pq.pop();
		int b = pq.top();
		pq.pop();
		pq.push(a + b * 2);

		answer++;
	}
	return answer;
}

File: PS/Programmers/더 맵게.py
import heapq


def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while True:
        a = heapq.heappop(scoville)
        if a >= K:
            return answer
        if len(scoville) == 0 and a < K:
            return -1
        b = heapq.heappop(scoville)
        heapq.heappush(scoville, a+(b+b))
        answer+=1
    return answer